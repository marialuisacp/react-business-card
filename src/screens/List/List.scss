@import '../../assets/styles/config';

#list-screen {
  .content.list {
    .carousel {
      position: relative;
      margin-top: -150px;
     
      .alphabet {
        height: 150px;
        display: block;
        text-align: center;

        .list-letters {
          display: flex;
          width: 390px;

          margin: 0 auto;
          position: relative;

          .letter {
            display: inline-block;
            border: none;
            background: transparent;
            width: 130px;
            height: 130px;
            outline: none;
            text-align: center;
            margin: 0 auto;
          }
          .letter span {
            color: #fff;
            font-size: 32px;
            font-weight: 300;
          }
        }
      }

      .slide {
        width: 130px;
        display: inline-block;        
        text-align: center;
        margin: 0 auto;
        .letter {
          width: 130px;
          height: 130px;
          border: none;
          outline: none;
          background: transparent;
        }
        .letter span {
          color: #fff;
          font-size: 32px;
          font-weight: 300;
        }
      }
    }
  }
}

@media (max-width: 1080px) {
  #list-screen {
    .content.list {
      .carousel {
        .alphabet {
          height: 120px;
          .list-letters {
            width: 100%;
            .letter {
              height: 110px;
            }
            .letter span{
              font-size: 28px;
            }
          }
        }
      }
    }
  }
}