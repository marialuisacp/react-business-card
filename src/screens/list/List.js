import React, { Component } from 'react';
import Header from '../../components/header/Header';

import { updateLetter } from '../../actions';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import './List.scss';
import '../../styles/styles.scss';
import '../../styles/lib/react-carousel.es.css';
import Alphabet from '../../components/alphabet/Alphabet';
import SwipeList from '../../components/swipeList/SwipeList';
import { dataPeople } from '../../utils';

class List extends Component {
  constructor(props) {
    super(props);
    this.state = {
      letters: ['', 'D', 'E', 'G', 'J', 'L', 'P'],
      data: dataPeople
    }
  }

  render() {
    const { currentLetter } = this.props;
    const { data } = this.state;
    const dataPeopleByLetter = {
      current: data[1],
      prev: data[0],
      next: data[2]
    }
    return (
      <div className='screen' id='list-screen'>
        <Header />
        <Alphabet letterCenter={currentLetter} />
        <SwipeList data={dataPeopleByLetter} />
      </div >
    );
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators({ updateLetter }, dispatch);

const mapStateToProps = store => ({
  currentLetter: store.letterState.currentLetter
});

export default connect(mapStateToProps, mapDispatchToProps)(List);