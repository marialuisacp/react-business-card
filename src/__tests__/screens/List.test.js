import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import List from '../../screens/List/List';
import { Provider } from 'react-redux'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('List', () => {
  test('renders without crashing given the required props', () => {
    const props = {
      letterCenter: 'D',
      dispatch: jest.fn(),
      direction: 1
    }

    const store = mockStore({
      currentLetter: 'D',
      direction: 0
    })

    const wrapper = shallow(<Provider store={store}>
      <List {...props} />
    </Provider>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})