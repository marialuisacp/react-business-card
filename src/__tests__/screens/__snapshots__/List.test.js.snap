// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders without crashing given the required props 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(List)
    direction={1}
    dispatch={[MockFunction]}
    letterCenter="D"
  />
</ContextProvider>
`;
